authui/src/password-policy.ts

```diff
@@ -1,106 +1,151 @@
+    "data-desc-" + score
+  );
 
-function checkPasswordPolicy(e: Event) {
-  const el = e.currentTarget as HTMLInputElement;
-  const value = el.value;
-  const els = document.querySelectorAll(".password-policy");
-  for (let i = 0; i < els.length; ++i) {
-    els[i].classList.remove("error-txt", "good-txt");
+  if (requiredMeter != null && strengthTarget != null) {
+    if (currentMeter.value >= requiredMeter.value) {
+      strengthTarget.classList.add("good-txt");
+    } else {
+      strengthTarget.classList.add("error-txt");
+    }
   }
-  checkPasswordLength(value, document.querySelector(".password-policy.length"));
-  checkPasswordUppercase(
-    value,
-    document.querySelector(".password-policy.uppercase")
-  );
-  checkPasswordLowercase(
-    value,
-    document.querySelector(".password-policy.lowercase")
-  );
-  checkPasswordDigit(value, document.querySelector(".password-policy.digit"));
-  checkPasswordSymbol(value, document.querySelector(".password-policy.symbol"));
-  checkPasswordStrength(value);
 }
 
-export function setupPasswordPolicy(): () => void {
-  const elems = document.querySelectorAll("[data-password-policy-password]");
-  for (let i = 0; i < elems.length; i++) {
-    elems[i].addEventListener("input", checkPasswordPolicy);
-  }
+export class PasswordPolicyController extends Controller {
+  static targets = [
+    "input",
+
+    "currentMeter",
+    "currentMeterDescription",
 
-  return () => {
-    for (let i = 0; i < elems.length; i++) {
-      elems[i].removeEventListener("input", checkPasswordPolicy);
+    "item",
+
+    "length",
+    "uppercase",
+    "lowercase",
+    "digit",
+    "symbol",
+    "strength",
+    "requiredMeter",
+  ];
+
+  declare inputTarget: HTMLInputElement;
+  declare currentMeterTarget: HTMLMeterElement;
+  declare currentMeterDescriptionTarget: HTMLElement;
+  declare itemTargets: HTMLElement[];
+  declare lengthTarget: HTMLElement | null | undefined;
+  declare uppercaseTarget: HTMLElement | null | undefined;
+  declare lowercaseTarget: HTMLElement | null | undefined;
+  declare digitTarget: HTMLElement | null | undefined;
+  declare symbolTarget: HTMLElement | null | undefined;
+  declare strengthTarget: HTMLElement | null | undefined;
+  declare requiredMeterTarget: HTMLMeterElement | null | undefined;
+
+  check() {
+    const value = this.inputTarget.value;
+    for (let i = 0; i < this.itemTargets.length; i++) {
+      this.itemTargets[i].classList.remove("error-txt", "good-txt");
     }
-  };
+    checkPasswordLength(value, this.lengthTarget);
+    checkPasswordUppercase(value, this.uppercaseTarget);
+    checkPasswordLowercase(value, this.lowercaseTarget);
+    checkPasswordDigit(value, this.digitTarget);
+    checkPasswordSymbol(value, this.symbolTarget);
+    checkPasswordStrength(
+      value,
+      this.currentMeterTarget,
+      this.currentMeterDescriptionTarget,
+      this.requiredMeterTarget,
+      this.strengthTarget
+    );
+  }
```

Comment:
Got following errors when some of the password policy is not enabled.
```
authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1 Error invoking action "input->password-policy#check"

Error: Missing target element "requiredMeter" for "password-policy" controller
    at a.get (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:44553)
    at a.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:1011089)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:7029)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:6587)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:1972)
```

Seems we should check the existence of the target before accessing by `this.has[Name]Target`

https://stimulus.hotwired.dev/reference/targets